### PR TITLE
Hidden webview

### PR DESCRIPTION
**Pull request by LiorAgnin**
## Proposed changes

- After signup successfully opening webview for create 3box account.
- Every switch community action open webview for join the user to the community.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: lib/logic/crypto.dart [hunk 1]
```diff
@@ -39,13 +39,11 @@ Future getBalance(accountAddress, tokenAddress) async {
   return balance;
 }
 
-Future sendNIS(address, amount, privateKey) async {
+Future sendNIS(toAccountAddress, amount) async {
```
Comment (leonprou) [inline lib/logic/crypto.dart]: can rename it to `sendToken`

File: lib/logic/wallet_logic.dart [hunk 2]
```diff
@@ -40,6 +40,7 @@ class WalletLogic {
 
       //Call funder
       var publicKey = await getPublickKey();
```
Comment (leonprou) [inline lib/logic/wallet_logic.dart:43]: It's not quite accurate cause actually it's Ethereum  account address and not public key (https://ethereum.stackexchange.com/questions/33171/ethereum-address-vs-public-key)

File: lib/modals/businesses.dart [hunk 3]
```diff
@@ -43,14 +43,26 @@ class BusinessList {
...
     final dynamic responseJson = json.decode(response.body);
     responseJson["data"].forEach((f) => businessList.add(new Business.fromJson(f)));
     print('Done Fetching businesses for commnuity: $communityAddress. length: ${businessList.length}');
     return businessList;
-  });
+  } else {
+    throw Exception('Failed to load business');
+  }
+
+  // return http.get(API_ROOT + "entities/" + communityAddress + "?type=business&withMetadata=true").then((response) {
```
Comment (leonprou) [inline lib/modals/businesses.dart]: I think you can delete this

File: lib/logic/crypto.dart [hunk 4]
```diff
@@ -54,12 +52,12 @@ Future sendNIS(address, amount, privateKey) async {
 
 //, EtherAmount.fromUnitAndValue(EtherUnit.gwei, 1)
   var getKittyFn = contract.findFunctionsByName("transfer").first;
```
Comment (leonprou) [inline lib/logic/crypto.dart]: you can also rename the variables, this examples is taken from cryptokitties :)

File: lib/modals/businesses.dart [hunk 5]
```diff
@@ -43,14 +43,26 @@ class BusinessList {
   }
 }
 
-Future<List<Business>> getBusinesses() async {
-  var communityAddress = await getCommunityAddress();
+Future<List<Business>> getBusinesses(String communityAddress) async {
+  // var communityAddress = await getCommunityAddress();
   print('Fetching businesses for commnuity: $communityAddress');
-  return http.get(API_ROOT + "entities/" + communityAddress + "?type=business&withMetadata=true").then((response) {
+  final response = await http.get(API_ROOT + "entities/" + communityAddress + "?type=business&withMetadata=true");
```
Comment (leonprou) [inline lib/modals/businesses.dart:48]: can you recheck that `withMetadata` is needed with the entities endpoint? Now we save 3box data in `profiles` so maybe this flag is redundant

File: lib/redux/actions/wallet_actions.dart [hunk 6]
```diff
@@ -10,6 +11,22 @@ ThunkAction initWalletCall() {
...
+  return (Store store) async {
+    loadBusinesses(store);
+  };
+}
+
+Future loadBusinesses(Store store) async {
+  print('loadBusinesses');
+  final String communityAddress = store.state.walletState.communityAddress;
+  store.dispatch(new StartFeatching());
+  var businesseses = await getBusinesses(communityAddress);
```
Comment (leonprou) [inline lib/redux/actions/wallet_actions.dart:27]: Fetching*

File: lib/screens/web.dart [hunk 7]
```diff
@@ -130,10 +143,17 @@ document.head.appendChild(script1);
       ),
       backgroundColor: const Color(0xFFF8F8F8),
       body: new WebviewScaffold(
-        url: "https://communities-qa.cln.network",
+        url: "http://0.0.0.0:9000/view/join/0x3e6Fcd5770020A237D14735A7EFC899B6245eeb0",
```
Comment (leonprou) [inline lib/screens/web.dart]: change to QA :)
also the community address is static (if used)

**Review by leonprou — APPROVED** [review]

**Comment by leonprou** [comment]
This looks much better organised with redux, so good job!
Wrote you some suggestions for change, please go over. Also on some point (maybe now) we need to go over and delete all most of commented code..
One more question is about mnemonic and private key. As I understand, they are stores in storage and not leaving the device, right?

**Comment by LiorAgnin** [comment]
> This looks much better organised with redux, so good job!
> Wrote you some suggestions for change, please go over. Also on some point (maybe now) we need to go over and delete all most of commented code..
> One more question is about mnemonic and private key. As I understand, they are stores in storage and not leaving the device, right?

File: lib/modals/businesses.dart (hunk 5 above)
Comment (LiorAgnin) [reply lib/modals/businesses.dart:48]: I'll change it later on, after 3box will be fully integrate

**Comment by creatix** [comment]
> This looks much better organised with redux, so good job!
> Wrote you some suggestions for change, please go over. Also on some point (maybe now) we need to go over and delete all most of commented code..
> One more question is about mnemonic and private key. As I understand, they are stores in storage and not leaving the device, right? **Yes**

Thanks, but there still more work to be done organizing stuff.. 
Where are the suggestions? i don't see them..

**Pull request closed without merging**